// react-h5-audio-player/lib/utils.js
"use strict";

exports.__esModule = true;
exports.throttle = throttle;
exports.getDisplayTimeBySeconds = exports.getPosX = exports.getMainLayoutClassName = void 0;

var getMainLayoutClassName = function getMainLayoutClassName(layout) {
  switch (layout) {
    case 'stacked':
      return 'rhap_stacked';

    case 'stacked-reverse':
      return 'rhap_stacked-reverse';

    case 'horizontal':
      return 'rhap_horizontal';

    case 'horizontal-reverse':
      return 'rhap_horizontal-reverse';

    default:
      return 'rhap_stacked';
  }
};

exports.getMainLayoutClassName = getMainLayoutClassName;

var getPosX = function getPosX(event) {
  if (event instanceof MouseEvent) {
    return event.pageX || event.clientX;
  } else {
    return event.touches[0].pageX || event.touches[0].clientX;
  }
};

exports.getPosX = getPosX;

var addHeadingZero = function addHeadingZero(num) {
  return num > 9 ? num.toString() : "0" + num;
};

var getDisplayTimeBySeconds = function getDisplayTimeBySeconds(seconds) {
  if (!isFinite(seconds)) {
    return '00:00';
  }

  var min = addHeadingZero(Math.floor(seconds / 60));
  var sec = addHeadingZero(Math.floor(seconds % 60));
  return min + ":" + sec;
};

exports.getDisplayTimeBySeconds = getDisplayTimeBySeconds;

function throttle(func, limit) {
  var inThrottle = false;
  return function (arg) {
    if (!inThrottle) {
      func(arg);
      inThrottle = true;
      setTimeout(function () {
        return inThrottle = false;
      }, limit);
    }
  };
}
